presto-cache/src/test/java/com/facebook/presto/cache/alluxio/TestAlluxioCachingFileSystem.java

```diff
@@ -199,8 +224,66 @@ public void testCreationFailure()
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, 100);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, PAGE_SIZE);
+            validateBuffer(data, pageOffset + 10, buffer, 0, 100);
+
+            // read within the cached page
+            resetBaseline();
+            assertEquals(readFully(fileSystem, pageOffset + 20, buffer, 0, 90), 90);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE, 0);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, 90);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, PAGE_SIZE);
+            validateBuffer(data, pageOffset + 20, buffer, 0, 90);
+
+            // read partially after the range of the cache
+            resetBaseline();
+            assertEquals(readFully(fileSystem, pageOffset + PAGE_SIZE - 10, buffer, 0, 100), 100);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE, 0);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, 100);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, 2 * PAGE_SIZE);
+            validateBuffer(data, pageOffset + PAGE_SIZE - 10, buffer, 0, 100);
+
+            // read partially before the range of the cache
+            resetBaseline();
+            assertEquals(readFully(fileSystem, pageOffset - 10, buffer, 10, 50), 50);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE, 0);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, 50);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, 2 * PAGE_SIZE);
+            validateBuffer(data, pageOffset - 10, buffer, 10, 50);
+
+            // skip one page
+            resetBaseline();
+            assertEquals(readFully(fileSystem, pageOffset + PAGE_SIZE * 3, buffer, 40, 50), 50);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE, 0);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, 50);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, PAGE_SIZE);
+            validateBuffer(data, pageOffset + PAGE_SIZE * 3, buffer, 40, 50);
+
+            // read between cached pages
+            resetBaseline();
+            assertEquals(readFully(fileSystem, pageOffset + PAGE_SIZE * 2 - 10, buffer, 400, PAGE_SIZE + 20), PAGE_SIZE + 20);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_CACHE, 0);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_REQUESTED_EXTERNAL, PAGE_SIZE + 20);
+            checkMetrics(MetricKey.CLIENT_CACHE_BYTES_READ_EXTERNAL, 3 * PAGE_SIZE);
+            validateBuffer(data, pageOffset + PAGE_SIZE * 2 - 10, buffer, 400, PAGE_SIZE + 20);
+
+            state = MetricsSystem.counter(MetricKey.CLIENT_CACHE_STATE.getName()).getCount();
+            assertTrue(state == CacheManager.State.READ_ONLY.getValue()
+                    || state == CacheManager.State.NOT_IN_USE.getValue());
```

Comment:
same

Reply:
done. Is there any documented java style guideline for the codebase?